scripts/bfasst/compare/waveform.py

```diff
@@ -0,0 +1,355 @@
+        print("Module 1: " + IMPL_NAME)
+        print("Module 2: " + REVERSED_NAME)
+
+        self.generate_files(design)
+
+        if(self.runTest(design)):
+            return self.success_status
+        else:
+            return Status(CompareStatus.NOT_EQUIVALENT)
+
+    #Returns the sizes of input, output, and the total list.
+    def inputNum(self):
+        return(len(data["inputList"]))
+
+    def outputNum(self):
+        return(len(data["outputList"]))
+
+    def totalNum(self):
+        return(len(data["inputList"]) + len(data["outputList"]))
+
+    def fix_file(self, PATH, fileName, isReversed):
+        fin = open(PATH + fileName + ".v", "r")
+        fileData = fin.read()
+        if(isReversed):
+            fileData = fileData.replace("module top(", "module " + fileName + "(")
+        else:
+            fileData = fileData.replace("module " + fileName[0:len(fileName)-5] + "\n", "module " + fileName)
+        fin.close()
+        fin = open(PATH + fileName + ".v", "w")
+        fin.write(fileData)
+        fin.close()
+
+    def parse(self, file):
+        netlist = sdn.parse(file)
+        library = netlist.libraries[0]
+        definition = library.definitions[0] #CHANGE THIS FOR MULTIPLE-MODULE DESIGNS.
+
+        i = 0
+        while(i < len(definition.ports)):
+            if(str(definition.ports[i].direction) == "Direction.OUT"):
+                data["outputList"].append(definition.ports[i].name)
+                data["totalList"].append(definition.ports[i].name)
+                data["oBitsList"].append(len(definition.ports[i].pins)-1)
+            elif (str(definition.ports[i].direction) == "Direction.IN"):
+                data["inputList"].append(definition.ports[i].name)
+                data["totalList"].append(definition.ports[i].name)
+                data["iBitsList"].append(len(definition.ports[i].pins)-1)
+            i = i + 1
+
+    def parseReversed(self, path, file):
+        file = open(file)
+        if(exists(path + "test.v")):
+            os.remove(path + "test.v")
+        newFile = open(path + "test.v", "x")
+        i=0
+
+        for line in file:
+            if "module" in line:
+                newFile.write(line)
+            elif "input" in line:
+                newFile.write(line)
+            elif "output" in line:
+                newFile.write(line)
+            else:
+                if (i == 0):
+                    i = 1
+                    newFile.write(line)
+        file.close()
+        newFile.close()
+        self.parse(newFile.name)
+        os.remove(path + "test.v")
+
+    def generateFirstTestbench(self, tb, line, tests, fileName):
+        if "TB_NAME;" in line:
+            line = ("module " + fileName[0] + "_tb;")
+            tb.write(line)
+            line = "\n"
+        
+        if "TB_NAME)" in line:
+            line=("    $dumpvars(0," + fileName[0] + "_tb);\n")
+        
+        if "INPUTS" in line:
+            i = 0
+            while(i < self.inputNum()):
```

Comment:
Don't use loops like this in python.  Either iterate through the object list, or if you need a numbered iterator use:
`for i in range(self.inputNum())`